### PR TITLE
use isort to sort python imports

### LINKED ISSUE #1622: Consider isort for import ordering matching our STYLE
https://github.com/PyCQA/isort

### PR DESCRIPTION
**Pull request by mwartell**
Fixes #1622 

Running `isort *` with the "black" profile does do a nice job of reordering and regrouping imports. More importantly, it doesn't appear to have done anything broken or ugly.  Also updated:

 - pyproject.toml for isort profile 
 - pre-commit.sh for isort run
 - style.md

Because this touches 130+ files with format only changes, I am targeting this request for `develop` as there is no cause to put it to `rc0.16.1`.

### CODE REVIEW

**Review by davidslater — COMMENTED** [review]
It generally looks good. Is it configurable at all? it looks like it orders based on the final imported name, not including the prefix, which is a bit counterintuitive. It also groups `import` separate from `from`.

**Comment by mwartell** [comment]
I think that https://pycqa.github.io/isort/docs/configuration/options.html#force-sort-within-sections covers both of your concerns. I know it stops the separation of `import` and `from`, and think that it may fix the ordering "not including prefix".

File: armory/__init__.py [hunk 1]
```diff
@@ -2,8 +2,7 @@
 Adversarial Robustness Evaluation Test Bed
 """
 from armory.logs import log
```
Comment (mwartell) [inline armory/__init__.py:4]: This pull request is vastly easier to read with the Split Diff View under the gear icon up at top.

File: armory/art_experimental/attacks/snr_pgd.py [hunk 2]
```diff
@@ -308,10 +308,9 @@ def _compute_perturbation(self, x, y, mask):
                      perturbed.
         :return: Perturbations.
         """
-        import torch  # lgtm [py/repeated-import]
-
         # Get gradient wrt loss; invert it if attack is targeted
         import art
```
[the remark this answers is not in the document]
Comment (davidslater) [reply armory/art_experimental/attacks/snr_pgd.py:312]: Is there a way to disable it on a per-line basis?

[the remark this answers is not in the document]
Comment (mwartell) [reply armory/art_experimental/attacks/snr_pgd.py:312]: Yes, `# isort:skip`, as I have applied.

**Comment by mwartell** [comment]
Supplanted by #1789

**Pull request closed without merging**